mlflow/utils/thread_utils.py

```diff
@@ -45,4 +46,10 @@ def set(self, value):
         Set a value for the thread-local variable.
         """
         self.thread_local.value = (value, os.getpid())
-        self._value_dict[threading.currentThread().ident] = value
+        self.__global_thread_values[threading.currentThread().ident] = value
```

Comment:
```suggestion
        self.__global_thread_values[threading.currentThread().get_ident()] = value
```

The official python doc doesn't mention an `ident` attribute. 

https://docs.python.org/3/library/threading.html

Let's use something that's mentioned for safety.